eth/tracers/native/call.go

```diff
@@ -92,13 +94,20 @@ func (t *callTracer) CaptureStart(env *vm.EVM, from common.Address, to common.Ad
 // CaptureEnd is called after the call finishes to finalize the tracing.
 func (t *callTracer) CaptureEnd(output []byte, gasUsed uint64, _ time.Duration, err error) {
 	t.callstack[0].GasUsed = uintToHex(gasUsed)
-	if err != nil {
-		t.callstack[0].Error = err.Error()
-		if err.Error() == "execution reverted" && len(output) > 0 {
-			t.callstack[0].Output = bytesToHex(output)
-		}
-	} else {
+	if err == nil {
 		t.callstack[0].Output = bytesToHex(output)
+		return
+	}
+	t.callstack[0].Error = err.Error()
+	if err != vm.ErrExecutionReverted || len(output) == 0 {
+		return
+	}
+	t.callstack[0].Output = bytesToHex(output)
+	if len(output) < 5 {
+		return
+	}
+	if unpacked, err := abi.UnpackRevert(output); err == nil {
```

Comment:
Feels strange to just ignore the error if it happens. Also to just silently not do anything if the `len(output)` is too short. IMHO we should at least log a warning to the user?

Reply:
I gave this a try, and it seems it's not so uncommon for this unpack to fail. Because some contracts have custom revert data instead of the string messages that solidity returns by default.

Reply:
I agree, there's nothing that _mandates_ that a contract's revert message is ABI-formatted. If it's not, then I think it's fine to not put anything into the "revert-message according to abi specificaton" (and yes, ignore the error)